btree/btree.c

```diff
@@ -3019,6 +3072,22 @@ void m0_btree_put_credit(const struct m0_btree  *tree,
 	m0_be_tx_credit_mac(accum, &cred, nr);
 }
 
+/**
+ * This function will calculate credits required to perform @nr delete KV
+ * operations and it will add those credits to @accum.
+ */
+void m0_btree_del_credit(const struct m0_btree  *tree,
+			 m0_bcount_t             nr,
```

Comment:
Do we ever have nr > 1 ? As far as the del_kv_tick routine is concerned I think you will be doing one record delete at a time. While having nr is fine the call to m0_be_tx_credit_mac at the end of this function will consume a lot of credits.

Reply:
Yes, nr > 1 , for /cas/ctg_store.c: ctg_meta_delete_credit(), /be/extmap.c: m0_be_emap_credit , etc.